### PR TITLE
Feature/orca 614 Migrate internal reconciliation db code to graphql

### PR DESCRIPTION
**Pull request by AndrewDorn**
## Summary of Changes

Please write a sentence or two summarizing the proposed change.

Addresses [ORCA-614: Migrate internal reconciliation db code to graphql](https://bugs.earthdata.nasa.gov/browse/ORCA-614)

## Changes

* Copied internal reconciliation database code to GQL.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Partially tested locally. Full tests will come with upcoming tasks.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
    - [x] Unit tests
- [x] New and existing unit tests pass locally with my changes
    - [x] Automated tests passing (if not fork)
- [x] Any dependent changes have been merged and published in downstream modules
- [x] My code has passed security scanning
    - [x] Snyk
    - [x] git-secrets

### CODE REVIEW

**Review by rizbihassan — COMMENTED** [review]
Lots of stuff and migration. Looks good to me from what I understand about graphql.
What happens to the api gateway for internal reconciliation now? In other words, does anything change on how we access the api?

File: modules/orca/main.tf [hunk 1]
```diff
@@ -260,11 +260,12 @@ module "orca_graphql_1" {
   ## ORCA Variables
   ## --------------------------
   ## REQUIRED
-  db_connect_info_secret_arn      = module.orca_secretsmanager.secretsmanager_arn
-  ecs_cluster_id                  = module.orca_ecs.ecs_cluster_id
-  gql_ecs_task_execution_role_arn = module.orca_graphql_0.gql_ecs_task_execution_role_arn
-  gql_ecs_task_execution_role_id  = module.orca_graphql_0.gql_ecs_task_execution_role_id
-  gql_tasks_role_arn              = module.orca_graphql_0.gql_tasks_role_arn
+  db_connect_info_secret_arn       = module.orca_secretsmanager.secretsmanager_arn
+  s3_access_credentials_secret_arn = module.orca_secretsmanager.s3_access_credentials_secret_arn
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply modules/orca/main.tf:264]: Same key. No migration needed.

**Comment by AndrewDorn** [comment]
> Lots of stuff and migration. Looks good to me from what I understand about graphql. What happens to the api gateway for internal reconciliation now? In other words, does anything change on how we access the api?

Not from this. These endpoints are internal, and are not exposed via API Gateway.

**Review by rizbihassan — CHANGES_REQUESTED** [review]
might want to update changelog.md as well

**Review by rizbihassan — APPROVED** [review]
looks good. Once this is merged, I will pull in the changes and fix the conflicts.

**Pull request merged**